### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by amrro**
## Linear Ticket

[PRO-436 - Migrate Nx to latest version](https://linear.app/govrn/issue/PRO-436/migrate-nx-to-latest-version) 

## Description
Update all nx plugins besides dependencies.
updating react dependencies is excluded.

### CODE REVIEW

File: apps/protocol-frontend/src/components/ContributionDetailShell.tsx [hunk 1]
```diff
@@ -76,7 +76,7 @@ const ContributionDetailShell = ({
                   : `${contribution?.status.name} 👀`}
               </Badge>
               <Text fontSize="sm" color="gray.600">
-                {contribution?.date_of_engagement}
+                {`${contribution?.date_of_engagement}`}
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/ContributionDetailShell.tsx:79]: Why do we need this change?

File: apps/protocol-frontend/src/components/CreateWaitlistUserForm.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { CreateUserFormValues } from '../types/forms';
 
 const useYupValidationResolver = (userValidationSchema: any) =>
   useCallback(
-    async data => {
+    async (data: any) => {
```
Comment (alexkeating) [inline apps/protocol-frontend/src/components/CreateWaitlistUserForm.tsx:15]: Is it possible to use a more specific type?

File: libs/protocol-client/src/lib/client/mint-event-example.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "eventFragment": {
```
Comment (alexkeating) [inline libs/protocol-client/src/lib/client/mint-event-example.json:2]: Is this being used anywhere?

File: apps/protocol-frontend/src/components/ContributionDetailShell.tsx (hunk 1 above)
Comment (amrro) [reply apps/protocol-frontend/src/components/ContributionDetailShell.tsx:79]: It's throwing the following error: 
`TS2322: Type 'string | Date' is not assignable to type 'ReactNode'`

**Pull request merged**